### DISCUSSION TITLE
Configuring IP pools

### DISCUSSION THREAD

**Discussion by roshansingh** [postalserver/postal#2752] [Help with using Postal]
I bought a VPS from Contabo and they have given me two IP addresses on the same ethernet port (eth0). When I run `ip a` command I can see both the IP addresses:

```
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether xxxxxxx ff:ff:ff:ff:ff:ff
    altname enp0s18
    altname ens18
    inet 217.xx.xx.xx/20 brd 217.xx.xx.xxx global eth0
       valid_lft forever preferred_lft forever
    inet 31.xx.xx.xxx/21 brd 31.xx.xx.xxx scope global eth0
       valid_lft forever preferred_lft forever
```
After that i configured the second IP (31.xx.xx.xx) as a new IP pool and assigned it to the organization. It is still sending emails from the first IP (217.xx.xx.xx). I used show original message in gmail to confirm.

I want to use both the IP addresses to deliver emails.

**Comment by Pacerino** [comment]
I think this is related to my issue too. I’ve resolved this with Ip rules. See https://github.com/postalserver/postal/discussions/2728

**Reply by Pacerino** [reply]
At the from address I’ve added the domains. example.org
mydomain.com
And so on

**Reply by roshansingh** [reply]
@Pacerino can you explain your steps. I looked at your discussion but am not able to follow.

**Reply by Pacerino** [reply]
1. Add your IP Pool
2. Assign it to the Organisation and to the server
3. Go to Mailserver -> Settings -> IP Rules
4. Create a new IP Rule
5. To the "From Address" add your domains you want to use the IP Pool. I've added all my organisation domains
6. choose your ip pool

In Postal there is the function `ip_pool_for_message` which assigns an IP pool to each message, only then is the pool taken. 
https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/app/models/server.rb#L277C1-L290C6

After i've made these changes/settings, my pool started to work.

![image](https://github.com/postalserver/postal/assets/22194063/5b505f71-4e58-4bf2-a2b4-f1a65bcc2400)
